marathon_acme/tests/test_sse_protocol.py

```diff
@@ -13,254 +15,268 @@ def loseConnection(self):
+        protocol.dataReceived(b'event:test2\r\n')
+        protocol.dataReceived(b'data:world\r\n\r\n')
 
-        assert_that(self.messages, Equals([
+        assert_that(messages, Equals([
             ('test1', 'hello'),
             ('test2', 'world')
         ]))
 
-    def test_id_ignored(self):
+    def test_id_ignored(self, protocol, messages):
         """
         When the id field is included in an event, it should be ignored.
         """
-        self.protocol.dataReceived(b'data:hello\r\n')
-        self.protocol.dataReceived(b'id:123\r\n\r\n')
+        protocol.dataReceived(b'data:hello\r\n')
+        protocol.dataReceived(b'id:123\r\n\r\n')
 
-        assert_that(self.messages, Equals([('message', 'hello')]))
+        assert_that(messages, Equals([('message', 'hello')]))
 
-    def test_retry_ignored(self):
+    def test_retry_ignored(self, protocol, messages):
         """
         When the retry field is included in an event, it should be ignored.
         """
-        self.protocol.dataReceived(b'data:hello\r\n')
-        self.protocol.dataReceived(b'retry:123\r\n\r\n')
+        protocol.dataReceived(b'data:hello\r\n')
+        protocol.dataReceived(b'retry:123\r\n\r\n')
 
-        assert_that(self.messages, Equals([('message', 'hello')]))
+        assert_that(messages, Equals([('message', 'hello')]))
 
-    def test_unknown_field_ignored(self):
+    def test_unknown_field_ignored(self, protocol, messages):
         """
         When an unknown field is included in an event, it should be ignored.
         """
-        self.protocol.dataReceived(b'data:hello\r\n')
-        self.protocol.dataReceived(b'somefield:123\r\n\r\n')
+        protocol.dataReceived(b'data:hello\r\n')
+        protocol.dataReceived(b'somefield:123\r\n\r\n')
 
-        assert_that(self.messages, Equals([('message', 'hello')]))
+        assert_that(messages, Equals([('message', 'hello')]))
 
-    def test_leading_colon_ignored(self):
+    def test_leading_colon_ignored(self, protocol, messages):
         """
         When a line is received starting with a ':' character, the line should
         be ignored.
         """
-        self.protocol.dataReceived(b'data:hello\r\n')
-        self.protocol.dataReceived(b':123abc\r\n\r\n')
+        protocol.dataReceived(b'data:hello\r\n')
+        protocol.dataReceived(b':123abc\r\n\r\n')
 
-        assert_that(self.messages, Equals([('message', 'hello')]))
+        assert_that(messages, Equals([('message', 'hello')]))
 
-    def test_missing_colon(self):
+    def test_missing_colon(self, protocol, messages):
         """
         When a line is received that doesn't contain a ':' character, the whole
         line should be treated as the field and the value should be an empty
         string.
         """
-        self.protocol.dataReceived(b'data\r\n')
-        self.protocol.dataReceived(b'data:hello\r\n\r\n')
+        protocol.dataReceived(b'data\r\n')
+        protocol.dataReceived(b'data:hello\r\n\r\n')
```

Comment:
The docstring doesn't give me a very clear idea of the intended behaviour here. Is there an empty `data` field followed by a non-empty `data` field with the values joined by a newline or something?